Formula/qt@6.rb

```diff
@@ -0,0 +1,172 @@
+class QtAT6 < Formula
+  desc "Cross-platform application and UI framework"
+  homepage "https://www.qt.io/"
+  url "https://download.qt.io/official_releases/qt/6.0/6.0.1/single/qt-everywhere-src-6.0.1.tar.xz"
+  sha256 "d13cfac103cd80b216cd2f73d0211dd6b1a1de2516911c89ce9c5ed14d9631a8"
+  license all_of: ["GFDL-1.3-only", "GPL-2.0-only", "GPL-3.0-only", "LGPL-2.1-only", "LGPL-3.0-only"]
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on xcode: :build
+
+  depends_on "assimp"
+  depends_on "dbus"
+  depends_on "double-conversion"
+  depends_on "freetype"
+  depends_on "glib"
+  depends_on "icu4c"
+  depends_on "jpeg"
+  depends_on "libb2"
+  depends_on "libpng"
+  depends_on "libproxy"
+  depends_on "llvm"
+  depends_on "pcre2"
+  depends_on "python@3.9"
+  depends_on "zstd"
+
+  uses_from_macos "cups"
+  uses_from_macos "krb5"
+  uses_from_macos "perl"
+  uses_from_macos "sqlite"
+  uses_from_macos "zlib"
+
+  patch :DATA
+
+  def install
+    ENV.append "LDFLAGS", "-framework GSS"
+    ENV.append "LDFLAGS", "-framework IOKit"
+
+    config_args = %W[
+      -release
+
+      -prefix #{HOMEBREW_PREFIX}
+      -extprefix #{prefix}
+
+      -plugindir share/qt/plugins
+      -qmldir share/qt/qml
+      -docdir share/doc/qt
+      -examplesdir share/qt/examples
+      -testsdir share/qt/tests
+
+      -libproxy
+      -no-feature-relocatable
+      -no-sql-odbc
+      -no-sql-psql
+      -no-sql-mysql
+      -system-sqlite
+    ]
+
+    cmake_args = std_cmake_args.reject { |s| s["CMAKE_INSTALL_PREFIX"] } + %W[
+      -DICU_ROOT=#{Formula["icu4c"].opt_prefix}
+      -DLLVM_INSTALL_DIR=#{Formula["llvm"].opt_prefix}
+
+      -DCMAKE_OSX_DEPLOYMENT_TARGET=#{MacOS.version}
+      -DINSTALL_MKSPECSDIR=share/qt/mkspecs
+      -DINSTALL_DESCRIPTIONSDIR=share/qt/modules
+      -DINSTALL_TRANSLATIONSDIR=share/qt/translations
+
+      -DFEATURE_pkg_config=ON
+    ]
+
+    system "./configure", *config_args, "--", *cmake_args
+    system "ninja"
+    system "ninja", "install"
+
+    # FIXME: This is required to enable installation of all configurations of
+    # a Qt built with Ninja Multi-Config until the following issue is fixed.
+    # https://gitlab.kitware.com/cmake/cmake/-/issues/20713
+    rm bin/"qt-cmake-private-install.cmake"
+
+    # Some config scripts will only find Qt in a "Frameworks" folder
+    frameworks.install_symlink Dir["#{lib}/*.framework"]
+
+    inreplace lib/"cmake/Qt6/qt.toolchain.cmake", /.*set.__qt_initial_.*/, ""
```

Comment:
Is this reported upstream?

Reply:
This fixme just copied from `qt-cmake-private-install.cmake`, now it is closed, qt left this file just keep compability for older cmake versions